src/network/serverpackethandler.cpp

```diff
@@ -857,11 +857,21 @@ void Server::handleCommand_PlayerItem(NetworkPacket* pkt)
 		return;
 	}
 
-	u16 item;
+	u16 old_wield_idx, new_wield_idx;
+	ItemStack old_itemstack, new_itemstack;
 
-	*pkt >> item;
+	// Get current wielded item and its idx
+	old_wield_idx = playersao->getWieldIndex();
+	playersao->getWieldedItem(&old_itemstack);
 
-	playersao->getPlayer()->setWieldIndex(item);
+	// Deserialize new wield idx from packet
+	*pkt >> new_wield_idx;
```

Comment:
The deserialization should happen before any actions

Reply:
Also since `new_wield_idx` is directly given to mods the server should clamp it to the allowed range.

Reply:
> The deserialization should happen before any actions

May I know why? The existing wield idx needs to be assigned to `old_wield_idx` first, and only then can the current wield idx be updated. If you insist, I could deserialize the new idx into a temp u16 before doing anything else, but it feels unnecessary to me, tbh.

I'll look into clamping the new idx, thanks for the heads-up.

Reply:
`Player::setWieldIndex` clamps the passed idx. I could just copy over the same code, but that'd result in code duplication (albeit only for two lines). Shall I go ahead? Maybe you have a better idea?

Reply:
> May I know why?

In this case it's irrelevant/minor, but you typically only want to do anything after grabbing all necessary data from the packet (and thus verifying that it's complete)

> Maybe you have a better idea?

You could call `getWieldIndex()` after setting the new index to get the clamped value.

Reply:
You're right, it makes sense to ensure the packet is intact before doing anything else. And yeah, `getWieldIndex()` would work too.

Reply:
Done!